ViewArticle.jsp

```diff
@@ -343,14 +343,68 @@ select * from user_ip where prefer=0 and title="${ title}";
 select * from comments where title="${ title}"
 </sql:query>
 
+<%---------------------------------------------------------------------- --%>
+
+
 <c:forEach items="${ result2.rows}" var="row">
 <div class="div3">
 <table>
 <tr><td><a> <c:out value="${ row.user}"></c:out> </a> </td></tr>
 <tr><td> <c:out value="${ row.comment}"></c:out> </td></tr>
 <tr><td><p class="time"> <c:out value="${ row.time}"></c:out> </p> </td></tr>
+<tr>
+<td>
+
+<!-- ............................................................... -->
+
+<sql:query var="result5" dataSource="${ db}">
+select * from comments_ip where prefer=1 and id="${ row.id}";
+</sql:query>
+
+<c:set var="q" value="0"></c:set>
+<c:forEach items="${ result5.rows}" var="row1">
+
+<c:set var="q" value="${ q+1}"></c:set>
+</c:forEach>
+
+<form action="/OOAD/Controller" method="post" class="popularity">
+<input type="hidden" name="page" value="comments_like">
+<input type="hidden" name="id" value="${ row.id}">
+<input type="hidden" name="title" value="${ title}">
+<button class="like"><i class="fa fa-thumbs-o-up"></i></button>
+<c:out value="${ q}"></c:out>
+</form>
+
+<!-- ............................................................... -->
+
+<sql:query var="result6" dataSource="${ db}">
+select * from comments_ip where prefer=0 and id="${ row.id}";
+</sql:query>
+
+<c:set var="r" value="0"></c:set>
+<c:forEach items="${ result6.rows}" var="row2">
+
+<c:set var="r" value="${ r+1}"></c:set>
+</c:forEach>
+
+
+<form action="/OOAD/Controller" method="post"  class="popularity">
+<input type="hidden" name="page" value="comments_dislike">
+<input type="hidden" name="id" value="${ row.id}">
+<input type="hidden" name="title" value="${ title}">
+<button class="dislike"><i class="fa fa-thumbs-o-down"></i></button>
+<c:out value="${ r}"></c:out>
+</form>

```

Comment:
@mjhmohamed 

What is purpose of this whole block?

Add a high-level description for it.

**Hui**